src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -975,6 +976,26 @@ private static string CreateQueryString(IDictionary<string, string> queryString)
             return String.Join("&", queryString.Select(kvp => kvp.Key + "=" + kvp.Value).ToArray());
         }
 
+        /// <summary>
+        /// Gets the query from a url and removes the query string from the url passed
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>the query string from the url passed and removes the query string from the url.
+        /// If a query string is not in the url it returns null and leaves the url untouched</returns>
+        private static string GetQueryStringFromUrl(ref string url)
+        {
+            if (url == null) throw new ArgumentNullException("url");
+            string queryString = null;
+            if (url.Contains("?"))
```

Comment:
This probably should not be needed -  you could Split and check the number of items in the returned array. As commented below - I think Split may not work in some cases so I would recommend finding the index of first `?` and splitting on the index (if not -1)